src/Orleans.Runtime/Core/InsideRuntimeClient.cs

```diff
@@ -669,16 +650,7 @@ public void ReceiveResponse(Message message)
             }
         }
 
-        public string CurrentActivationIdentity
-        {
-            get
-            {
-                if (RuntimeContext.Current == null) return this.HostedClient.ToString();
-
-                var currentActivation = GetCurrentGrainContext();
-                return currentActivation.Address.ToString();
-            }
-        }
+        public string CurrentActivationIdentity => RuntimeContext.CurrentGrainContext?.Address.ToString() ?? this.HostedClient.ToString();
```

Comment:
Shouldn't the value be caches?

Reply:
I think it's not worth it - this is only used in rare logs.